internal/sync/lifecycle.go

```diff
@@ -182,6 +184,16 @@ func (l *lifecycleOnce) Stop(f func() error) error {
 	return l.loadError()
 }
 
+// Started returns a channel that will close when the lifecycle starts.
+func (l *lifecycleOnce) Started() <-chan struct{} {
```

Comment:
Only the sender side of a channel can close.

https://play.golang.org/p/x-4QwERTqe

This is safe and simple. As Will points out, I need the channel so I can wait for it in a `select {}` block to race multiple conditions that might wake the goroutine, as in https://github.com/yarpc/yarpc-go/pull/1058.